### PR TITLE
fix: omit timestamp when creating logentry when no timestamp is provided 

### PR DESCRIPTION
**Pull request by capri-xiyue**
omit timestamp when creating logentry when no timestamp is provided, cloud logging will automatically fill in the timestamp field.

### CODE REVIEW

File: clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/CloudLoggingProcessor.java [hunk 1]
```diff
@@ -77,12 +78,15 @@ public AuditLogRequest process(AuditLogRequest auditLogRequest) throws LogProces
...
                       .setLast(operation.getLast())
-                      .build())
-              .setTimestamp(
-                  Instant.ofEpochSecond(
-                      auditLogRequest.getTimestamp().getSeconds(),
-                      auditLogRequest.getTimestamp().getNanos()))
-              .build();
+                      .build());
+
+      if (!(auditLogRequest.getTimestamp()).equals(Timestamp.getDefaultInstance())) {
```
Comment (yolocs) [inline clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/CloudLoggingProcessor.java:83]: Is the default instance null?

**Review by yolocs — APPROVED** [review]

File: clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/CloudLoggingProcessor.java (hunk 1 above)
Comment (sqin2019) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/CloudLoggingProcessor.java:83]: shoud use hasTimestamp() instead?

Comment (capri-xiyue) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/CloudLoggingProcessor.java:83]: default instance is not null. I tried to use null check but intellij indicates that timestamp will always not be null

Comment (capri-xiyue) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/CloudLoggingProcessor.java:83]: > shoud use hasTimestamp() instead?

I didn't know we have such method in Java protobuf timestamp.  This is more clear. Updated it.

**Review by sqin2019 — APPROVED** [review]

**Pull request merged**